X2WOTCCommunityPromotionScreen/Src/X2WOTCCommunityPromotionScreen/Classes/CPS_UIArmory_PromotionHero.uc

```diff
@@ -229,6 +233,24 @@ simulated function PopulateData()
 		}
 	}
 
+	// Begin Issue #84
+	`LOG(Unit.GetFullName() @ "uses large armory scale:" @ Unit.UseLargeArmoryScale(), bLog, 'PromotionScreen');
+	if (Unit.UseLargeArmoryScale())
+	{
+		`LOG("Adding offset to pawn locatin.", bLog, 'PromotionScreen');
+		PawnLocation = ActorPawn.Location;
+		if (bHasBrigadierRank)
+		{
+			PawnLocation.X += LargeUnitPawnOffset_Brigadier;
+		}
+		else
+		{
+			PawnLocation.X += LargeUnitPawnOffset;
+		}
+		XComUnitPawnNativeBase(ActorPawn).SetLocationNoCollisionCheck(PawnLocation);
```

Comment:
I'm not 100% sure if this is the correct place to update this -- conceptually this should probably happen when actually creating the pawn so that the `UIPawnMgr`'s internal info also has the correct info.

Also this makes `PopulateData` not idempotent, which seems dangerous to me -- if `PopulateData` is called multiple times (skimming the code, e.g. by purchasing more abilities) the Spark would move a bunch of units to +X multiple times...

Reply:
> I'm not 100% sure if this is the correct place to update this -- conceptually this should probably happen when actually creating the pawn so that the `UIPawnMgr`'s internal info also has the correct info.
> 
> Also this makes `PopulateData` not idempotent, which seems dangerous to me -- if `PopulateData` is called multiple times (skimming the code, e.g. by purchasing more abilities) the Spark would move a bunch of units to +X multiple times...

Good points, will adjust.